packages/gestalt/src/Button.js

```diff
@@ -54,6 +54,7 @@ export default function Button(props: Props) {
     [styles.block]: !inline,
   });
 
+  /* eslint-disable react/button-has-type */
```

Comment:
@pfarejowicz Any idea why we need to disable this even though we set it below? Is it because `type` is dynamic?

`type={type}`

Reply:
@christianvuerings exactly! the rule does not like us using the dynamic value `type={type}` even if it is Flow typed and prop typed to match the possible values. There's more discussion in https://github.com/yannickcr/eslint-plugin-react/issues/1555 and the recommendation is to fork the component and render two different JSX components but to me that feels like overkill, what are your thoughts? For reference:

> I'd say that the rule should be forcing a single, hardcoded type value - and that you should conditionally render two different elements if you want two types.

Reply:
Disabling it here seems fine - thanks for clarifying